src/backend/base/langflow/components/prototypes/UpdateData.py

```diff
@@ -1,40 +1,119 @@
-from langflow.custom import CustomComponent
+from typing import Any
+
+from langflow.custom import Component
+from langflow.inputs.inputs import IntInput, MessageTextInput, DictInput, DataInput, BoolInput
+from langflow.io import Output
+
+from langflow.field_typing.range_spec import RangeSpec
 from langflow.schema import Data
+from langflow.schema.dotdict import dotdict
 
 
-class UpdateDataComponent(CustomComponent):
-    display_name = "Update Data"
-    description = "Update Data with text-based key/value pairs, similar to updating a Python dictionary."
-    name = "UpdateData"
-
-    def build_config(self):
-        return {
-            "data": {
-                "display_name": "Data",
-                "info": "The record to update.",
-            },
-            "new_data": {
-                "display_name": "New Data",
-                "info": "The new data to update the record with.",
-                "input_types": ["Text"],
-            },
-        }
-
-    def build(
-        self,
-        data: Data,
-        new_data: dict,
-    ) -> Data:
-        """
-        Updates a record with new data.
+class UpdateOrAppendDataComponent(Component):
+    display_name: str = "Update or Append Data"
+    description: str = "Dynamically update or append data with a specified number of fields."
```

Comment:
suggest: `Dynamically update or append data with the specified fields` (just remove number)